scipy/stats/_hypotests.py

```diff
@@ -497,31 +508,33 @@ def _cdf_cvm(x, n=None):
     and 1, respectively. These are limitations of the approximation by Csörgő
     and Faraway (1996) implemented in this function.
     """
-    x = np.asarray(x)
+    xp = array_namespace(x) if xp is None else xp
+    x = xp.asarray(x)
+
     if n is None:
-        y = _cdf_cvm_inf(x)
+        y = _cdf_cvm_inf(x, xp=xp)
     else:
         # support of the test statistic is [12/n, n/3], see 1.1 in [2]
-        y = np.zeros_like(x, dtype='float')
+        y = xp.zeros_like(x, dtype=x.dtype)
```

Comment:
Does `zeros_like` not match `x`'s dtype anyway?
```suggestion
        y = xp.zeros_like(x)
```